### PR TITLE
chore(docs): update github action example in managing environment guide

### PR DESCRIPTION
**Pull request by sweatybridge**
## What kind of change does this PR introduce?

docs update

## What is the new behavior?

- Update docs to match suggested [GHA example](https://github.com/supabase/supabase-action-example).
- Update logflare setup docs to mention dataset location requirement.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/docs/pages/guides/cli/managing-environments.mdx [hunk 1]
```diff
@@ -251,21 +255,22 @@ on:
 
 jobs:
   deploy:
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
```
Comment (silentworks) [inline apps/docs/pages/guides/cli/managing-environments.mdx:258]: We should probably pin the version as we don't want a non LTS version which might have breaking changes break the CI/CD pipeline.

File: apps/docs/pages/guides/cli/managing-environments.mdx [hunk 2]
```diff
@@ -251,21 +255,22 @@ on:
...
 
     env:
       SUPABASE_ACCESS_TOKEN: ${{ secrets.SUPABASE_ACCESS_TOKEN }}
       SUPABASE_DB_PASSWORD: ${{ secrets.STAGING_DB_PASSWORD }}
-      STAGING_PROJECT_ID: abcdefghijklmnopqrst
+      SUPABASE_PROJECT_ID: ${{ secrets.STAGING_PROJECT_ID }}
```
Comment (silentworks) [inline apps/docs/pages/guides/cli/managing-environments.mdx:263]: Is this a breaking change? as in will `STAGING_PROJECT_ID` still work or will existing projects be broken due to this change?

**Review by Ziinc — APPROVED** [review]

File: apps/docs/pages/guides/getting-started/local-development.mdx [hunk 3]
```diff
@@ -429,7 +429,8 @@ For these changes to take effect, you need to run `supabase stop` and `supabase
 Local logs rely on the Supabase Analytics Server. This can be enabled via the CLI configuration, and requires a Google Cloud project and BigQuery access.
 
 <Admonition type="note">
-  The Google Cloud project must have billing enabled. Read more about this requirement
+  The Google Cloud project must have billing enabled. Its dataset location must also be in the US.
```
Comment (Ziinc) [inline apps/docs/pages/guides/getting-started/local-development.mdx:432]:
```suggestion
  The Google Cloud project must have billing enabled. The dataset created and managed by Analytics will be within the US.
```

**Review by Ziinc — CHANGES_REQUESTED** [review]
user should not need to manage the dataset location themselves.

File: apps/docs/pages/guides/cli/managing-environments.mdx (hunk 2 above)
Comment (sweatybridge) [reply apps/docs/pages/guides/cli/managing-environments.mdx:263]: This is to keep our docs up to date for onboarding new users. There's no breaking change as we don't change any existing GHA workflows that customers have setup.

**Review by Ziinc — APPROVED** [review]

File: apps/docs/pages/guides/cli/managing-environments.mdx (hunk 1 above)
Comment (sweatybridge) [reply apps/docs/pages/guides/cli/managing-environments.mdx:258]: That's very unlikely to happen as the cli binary is agnostic to ubuntu versions, ie. we don't ship separate binaries for 20.04 or 22.04.

Comment (sweatybridge) [reply apps/docs/pages/guides/cli/managing-environments.mdx:258]: Also our [CI test](https://github.com/supabase/cli/blob/main/.github/workflows/ci.yml#L49) is using `ubuntu-latest` so if something really breaks, we will be the first to know.

**Pull request merged**